CMakeLists.txt

```diff
@@ -484,21 +484,13 @@ set(IO_SOURCE
         src/IO/ADIOS/ADIOS2PreloadAttributes.cpp
         src/IO/InvalidatableFile.cpp)
 set(IO_ADIOS1_SEQUENTIAL_SOURCE
-        src/Error.cpp
-        src/auxiliary/Filesystem.cpp
-        src/ChunkInfo.cpp
         src/IO/ADIOS/CommonADIOS1IOHandler.cpp
-        src/IO/ADIOS/ADIOS1IOHandler.cpp
-        src/IO/IOTask.cpp)
+        src/IO/ADIOS/ADIOS1IOHandler.cpp)
 set(IO_ADIOS1_SOURCE
-        src/Error.cpp
-        src/auxiliary/Filesystem.cpp
-        src/ChunkInfo.cpp
         src/IO/ADIOS/CommonADIOS1IOHandler.cpp
-        src/IO/ADIOS/ParallelADIOS1IOHandler.cpp
-        src/IO/IOTask.cpp)
```

Comment:
I don't fully understand why these need to be part of the ADIOS1 backend libs. Shouldn't it be enough to have the compiled symbols in libopenPMD.so and link later?
I removed them for now to reduce the number of files that must not include `Error.hpp`. If there are any .cpp files that should be added back here, we can set `#define OPENPMD_ADIOS1_IMPLEMENTATION 1` at its head. Error.hpp will throw a compile-time error if it finds that macro.

Reply:
@ax3l can you have a look at this, please?

Reply:
I think we need the symbols in these libs because the openPMD-api adios1 wrapper libs are not depending on the main openPMD-api libs (but the other way around).

Afaik, that only manifests as an issue in macOS. Maybe we can build a mini-app in macOS CI on our installed lib to cover this.

Reply:
We should definitely introduce such a CI check, as the error symbol workaround in this PR introduces a reverse dependency, where the ADIOS1 backend needs symbols from the main library. 
In the end, we might need to restrict the workaround to non-Apple systems.
I'll push a commit that reverses the changes in here.